src/ui/components/mixins/notebook-snapshot.js

```diff
@@ -0,0 +1,114 @@
+import $ from 'zepto';
+
+function generateTaskForm(url) {
+    var taskForm = {
+        name: "Create a Notebook Entry",
+        hint: "Please select a Notebook",
+        sections: [{
+            rows: [
+                {
+                    name: 'Entry',
+                    key: 'entry',
+                    control: 'textarea',
+                    required: false,
+                    cssClass: "l-textarea-sm"
+                },
+                {
+                    name: 'Snap Preview',
+                    key:"snapPreview",
+                    control: "snap-view",
+                    cssClass: "l-textarea-sm",
+                    src: url
+                },
+                {
+                    name: 'Save in Notebook',
+                    key: 'notebook',
+                    control: 'locator',
+                    validate: validateLocation
+                }
+            ]
+        }]
+    };
+
+    var overlayModel = {
+        title: taskForm.name,
+        message: 'Notebook Snapshot',
+        structure: taskForm,
+        value: {'entry': ""}
+    };
+
+    function validateLocation(newParentObj) {
+        return newParentObj.model.type === 'notebook';
+    }
+
+    return overlayModel;
+}
+
+function saveSnapShot(dialogService, imageUrl, imageType, imageSize, embedObject) {
+    let taskForm = generateTaskForm(imageUrl);
+
+    dialogService.getDialogResponse(
+        'overlay-dialog',
+        taskForm,
+        () => taskForm.value
+    ).then(options => {
+        let snapshotObject = {
+            src: options.snapPreview || imageUrl,
+            type: imageType,
+            size: imageSize
+        };
+
+        options.notebook.useCapability('mutation', function (model) {
+            var date = Date.now();
+
+            model.entries.push({
+                id: 'entry-' + date,
+                createdOn: date,
+                text: options.entry,
+                embeds: [{
+                    name: embedObject.name,
+                    cssClass: embedObject.cssClass,
+                    type: embedObject.id,
+                    id: 'embed-' + date,
+                    createdOn: date,
+                    snapshot: snapshotObject
+                }]
+            });
+        });
+    });
+}
+
+export default {
+    inject: ['openmct', 'domainObject'],
+    methods: {
+        takeSnapshot() {
+            let exportImageService = this.openmct.$injector.get('exportImageService'),
+                dialogService = this.openmct.$injector.get('dialogService');
+
+            let type = this.openmct.types.get(this.domainObject.type),
+                embedObject = {
+                    id: this.domainObject.identifier.key,
+                    cssClass: type.cssClass,
+                    name: this.domainObject.name
+                };
+
+            let elementToSnap =
```

Comment:
It would be nice if we could remove the hard-coded selectors here, and have the target DOM node passed in as an argument. I wonder if we should make this a library function that accepts a DOM element, rather than a mixin? It might make it a bit more flexible. The function could then be `import`ed wherever it needs to be used, and the calling code could provide the element to snapshot.

Reply:
I changed it to accept a dom element instead of hardcoding it, i think that makes more sense also

Reply:
This is much more flexible, thanks! 

What are your thoughts on making this a JavaScript class, instead of a mixin? eg.

```
let snapshot = new NotebookSnapshot(openmct);
snapshot.capture(domainObject, domElement);
```
There's no Vue-specific code in here, and making it a class makes it even more flexible (eg. can reused from non-Vue code like actions). The refactoring would be trivial.

As a general rule I think we should use built-in language features before frameworks, where it makes sense to. For one it reduces the amount of code we need to refactor next time we change frameworks :) For another it allows us to reuse the code in other contexts.

Reply:
I changed it into a class